### PR TITLE
issue #24490 and issue #24491 - improve command line interaction

### PR DESCRIPTION
**Pull request by gilmoskowitz**

### CODE REVIEW

**Comment by gilmoskowitz** [comment]
@bendiy @xtpclark @andersdd @davidbeauchamp - OK, guys... tear this apart and see what else needs to be done for command line support. What have I missed?

File: loader/main.cpp [hunk 1]
```diff
@@ -21,30 +21,26 @@
...
+  QString passwd;
+  QString pkgfile;
+  QString port;
+  QString username;
+  XAbstractMessageHandler *handler;
+  bool    autoRunArg      = false;
+  bool    autoRunCheck    = false;
+  bool    debugpkg        = false;
+  bool    haveDatabaseURL = false;
+  bool    acceptDefaults  = false;
```
Comment (bendiy) [inline loader/main.cpp]: @gilmoskowitz Are these the command line flags?

File: loader/main.cpp [hunk 2]
```diff
@@ -54,159 +50,167 @@ int main(int argc, char* argv[])
...
+                 " [ -h hostname ]"
+                 " [ -p port ]"
+                 " [ -d databasename ]"
+                 " [ -U username | -username=username ]"
+                 " [ -passwd=databasePassword ]"
                  " [ -debug ]"
-                 " [ -file=updaterFile.gz ]"
-                 " [ -autorun ]",
+                 " [ -file=updaterFile.gz | -f updaterFile.gz ]"
+                 " [ -autorun [ -D ] ]",
```
Comment (bendiy) [inline loader/main.cpp]: `-help`. looks like these are.

**Comment by bendiy** [comment]
@gilmoskowitz 
Does this support ignoring the license prompt?
What happens when a rollback error occurs when running headless?
Those were the two big issues we have had with xtrans.
Is there a build I can test?

**Comment by andersdd** [comment]
@gilmoskowitz 
Command line permissions and sudo escalation?  Is that assumed?
I'm also probably better off testing a build to see what I can find

File: loader/main.cpp (hunk 1 above)
Comment (gilmoskowitz) [inline loader/main.cpp]: As you probably realized, these are local variables that represent whether certain command line options have been passed in. Obviously they start as `false` then get updated when parsing the command line.

File: loader/main.cpp (hunk 2 above)
Comment (gilmoskowitz) [inline loader/main.cpp]: Kinda - this is the help output string. It's distinct from the code that processes the command line, which is unfortunate but common.

**Comment by gilmoskowitz** [comment]
@andersdd

The `-autorun` option now forces command-line interaction. If you add `-D` (capital letter D) to `-autorun`, the default values for the prompts are accepted automatically. Thus the answer to your first question is, "Yes, this supports ignoring the license prompt."

If there's an error then the headless updater should return non-zero, allowing error detection when scripted. There are several unit-test directories in https://github.com/xtuple/updater/tree/master/examples that seemed to work properly.

Command line permissions and `sudo` are non-issues. This is a database application and so is subject to database privileges, not operating system privs.

**Pull request merged**